R/edgeThresh.R

```diff
@@ -0,0 +1,284 @@
+#' edge A → B). Finds the value δ (edge weight) for each random graph such that
+#' the maximum size of any component in the graph is Lmax, and returns the
+#' median of N δ values.
+#'
+#' @section Swapping Edges:
+#'     Bidirectional edges must only be swapped with each other, therefore the
+#'     swapping operation will first be applied to bidirectional edges,
+#'     Q * |E_b| times, where E_b is a bidirectional edge. Since all 4 incident
+#'     nodes in 2 bidirectional edges are receiving endpoints, one node will be
+#'     randomly selected from each edge, and a new edge is formed between the
+#'     selected as well as the remaining vertices.
+#'     Then Q * |E_u| unidirectional edge swaps take place.
+#'     At each unidirectional edge swap, only the receiving endpoints are
+#'     swapped e.g. (A → B, C → D) to (A → D, C → B).
+#'     In both cases, a swap should NOT result in:
+#'     Multiple edges: Possibly result of a bidirectional edge swap.
+#'     Self-edges: e.g. (A ↔ B, C ↔ A) to (A ↔ A, B ↔ C)
+#'     Disconnected components: Considering (A → B, C → D), if A and B are
+#'     in the same component, they should still be in the same component when
+#'     (A → D, C → B) This guarantees that the graph is still connected if
+#'     the starting graph was also connected.
+#'     ### What about (A → B, A → D) to (A → D, A → B)? Will this be
+#'     cosidered a swap? (counted towards Q * |E|) ###
+#'
+#' @section Maintaining Connected Components:
+#'     In order to ensure nodes maintain their components during permutation,
+#'     the components of the graph is measured initially, and the component
+#'     each node belongs to is stored in a separate data structure (hash table?).
+#'     After each swap, the new component of the sending endpoint (A in A → B)
+#'     is the same as the component the receiving endpoint belongs to. Thus
+#'     the component maintenance is acieved in 2 steps (one query and one
+#'     comparison) for each receiving node, and thus 4 * Q * |E| steps are
+#'     required.
+#'
+#'     ## Why not use a binary search style approach to find the minimum delta?
+#'     ## Can calculation of delta be parallelized?
+#' @section Incrementing δ and Lmax: ## How much should delta be incremented for efficieny as well as accuracy?##
+#'     After creating the permutated graph, the following procedure is run on
+#'     the graph for each Lmax starting from 2 to the Lmax given as an argument.
+#'     On each such run, and a small value for δ is chosen (0.01) , and
+#'     the strongly connected component discovery is done. If the size of the
+#'     biggest component after such run is smaller than Lmax, increment δ by
+#'     (0.01), and redo component disovery. As soon as the largest component has
+#'     size Lmax (for current run), δ is recorded, and Lmax is incremented by 1.
+#'     Once current Lmax is the same as the given Lmax, run the procedure on the
+#'     next graph.
```

Comment:
Your suggestion for a binary search is good - perhaps there are even better methods, essentially this is a root finding problem on a function that does not have continuous derivatives.